pkg/operator/apis/monitoring/v1alpha1/types_test.go

```diff
@@ -372,7 +405,12 @@ honor_timestamps: false
 scrape_interval: 10s
 scrape_timeout: 10s
 metrics_path: /metrics
-follow_redirects: false
+sample_limit: 1
+label_limit: 2
+label_name_length_limit: 3
+label_value_length_limit: 4
+proxy_url: ""
+follow_redirects: true
```

Comment:
Where does `follow_redirects` get set?

Reply:
It's through the DefaultHTTPClientConfig.
That's one of the things that's broken and why we manually have to set the DefaultHTTPClientConfig. It doesn't have `omitempty` set on the upstream struct. Hence when encoding it explicitly sets `follow_redirects: false` and then no longer defaults correctly to `true` when decoding again.